packages/components/src/systems/SignatureAttestation/SignatureAttestationModal.tsx

```diff
@@ -0,0 +1,252 @@
+import { ApolloClient } from '@apollo/client';
+import gql from 'graphql-tag';
+import { createEffect, createSignal, JSXElement, Ref, Show } from 'solid-js';
+import Button from '../../particles/Button';
+import Card from '../../particles/Card';
+import Spinner from '../../particles/Spinner';
+import Text from '../../particles/Text';
+import { Maybe } from 'graphql/jsutils/Maybe';
+import Icon from '../../particles/Icon';
+
+const GetCurrentUserSignatureAttestationStatus = gql`
+  query GetCurrentUserSignatureAttestationStatus {
+    me {
+      signatureAttestationStatus {
+        ... on NeedsSignatureAttestation {
+          version
+          content
+        }
+        ... on CompletedSignatureAttestation {
+          agreedAt
+          version
+        }
+        ... on NotApplicableSignatureAttestation {
+          reason
+        }
+      }
+    }
+  }
+`;
+
+export type GetCurrentUserSignatureAttestationStatusType = {
```

Comment:
any way we can generate these the same way we're doing for app?

Reply:
Not easily / quickly. I have a TODO item to go back and do these though.